plugin/cscope_maps.vim

```diff
@@ -41,12 +41,40 @@ if has("cscope")
     " if you want the reverse search order.
     set csto=0
 
-    " add any cscope database in current directory
-    if filereadable("cscope.out")
-        cs add cscope.out
-    " else add the database pointed to by environment variable
-    elseif $CSCOPE_DB != ""
+    " cscope seems to not index files in symlinked directories.
+    " However, one can build the cscope.files (list of files for cscope to index)
+    " via a command like:
+    " `find -L [your searched folder] -name [your searched pattern]
+    "   -exec realpath {} \; >> cscope.files`
+    "
+    " Note that it is prudent to use absolute file paths, hence the use of realpath.
+    "
+    " Now the trouble is getting vim to try to find a cscope.out file
+    " going recursively up through directories, but going back up through
+    " them via any symlinks you took to get to your CWD in the shell.
+    " The following codes solves this issue.
+
+    if $CSCOPE_DB != ""
         cs add $CSCOPE_DB
+    else
+        " Get all parts of out current path
+        let dirs = split($PWD, '/')
```

Comment:
Is it possible to simplify the whole algorithm somehow? Get rid of all loops, to make it O(1) instead of O(n)? Again, I probably don't understand what the whole issue is about. From my point of view, we already know cscope.out file location, and it doesn't have a symlink in its path (or has it?). So why 'cs add' is gonna fail?